File: app/src/main/java/com/tasomaniac/openwith/preferred/PreferredAppsAdapter.java
package com.tasomaniac.openwith.preferred;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import com.tasomaniac.openwith.HeaderAdapter;
import com.tasomaniac.openwith.R;
import com.tasomaniac.openwith.resolver.ResolveListAdapter;
import com.tasomaniac.openwith.util.Utils;

class PreferredAppsAdapter extends HeaderAdapter {

    private final ResolveListAdapter innerAdapter;

    PreferredAppsAdapter(ResolveListAdapter innerAdapter) {
        super(innerAdapter, R.layout.preferred_header);
        this.innerAdapter = innerAdapter;
        innerAdapter.setDisplayExtendedInfo(true);
    }

    @Override
    protected void onBindHeaderViewHolder(HeaderViewHolder holder) {
        if (innerAdapter.getItemCount() == 0) {
            holder.setText(R.string.desc_preferred_empty);
        } else {
            holder.setText(R.string.desc_preferred);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return holderWithMinHeight(super.onCreateViewHolder(parent, viewType));
    }

    private RecyclerView.ViewHolder holderWithMinHeight(RecyclerView.ViewHolder viewHolder) {
        viewHolder.itemView.setMinimumHeight(Utils.dpToPx(viewHolder.itemView.getResources(), 72));
        return viewHolder;
    }

}

File: app/src/main/java/com/tasomaniac/openwith/preferred/PreferredAppsAdapter.kt
package com.tasomaniac.openwith.preferred

import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import com.tasomaniac.openwith.extensions.withMinHeight
import com.tasomaniac.openwith.resolver.ApplicationViewHolder
import com.tasomaniac.openwith.resolver.DiffUtilsCallback
import com.tasomaniac.openwith.resolver.DisplayActivityInfo
import com.tasomaniac.openwith.resolver.ItemClickListener
import javax.inject.Inject

class PreferredAppsAdapter @Inject constructor(
    private val viewHolderFactory: ApplicationViewHolder.Factory
) : ListAdapter<DisplayActivityInfo, ApplicationViewHolder>(DiffUtilsCallback) {

    var itemClickListener: ItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        viewHolderFactory.createWith(parent, displaySubtext = true).withMinHeight()

    override fun onBindViewHolder(holder: ApplicationViewHolder, position: Int) {
        holder.bind(getItem(position), itemClickListener)
    }
}
